File: java/trunk/TetramBDEJB/ejbModule/org/tetram/bdtheque/model/entity/AuteurSerie.java
package org.tetram.bdtheque.model.entity;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.DiscriminatorColumn;
import javax.persistence.DiscriminatorType;
import javax.persistence.Entity;
import javax.persistence.Enumerated;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name = "metier", discriminatorType = DiscriminatorType.STRING)
@Table(name = "auteurs_series")
public class AuteurSerie implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	@GeneratedValue
	@Id
	@Column(name = "id_auteur_series")
	private String id;

	@ManyToOne(optional = false, fetch = FetchType.LAZY)
	@JoinColumn(name = "id_personne")
	private Auteur auteur;
	@ManyToOne(optional = false, fetch = FetchType.LAZY)
	@JoinColumn(name = "id_serie")
	private Serie serie;

	@Enumerated
	@Column(nullable = false, insertable = false, updatable = false)
	private MetierAuteur metier;

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public Auteur getAuteur() {
		return auteur;
	}

	public void setAuteur(Auteur auteur) {
		this.auteur = auteur;
	}

	public Serie getSerie() {
		return serie;
	}

	public void setSerie(Serie serie) {
		this.serie = serie;
	}

	public MetierAuteur getMetier() {
		return metier;
	}

	public void setMetier(MetierAuteur metier) {
		this.metier = metier;
	}

}

File: bdtheque/php/trunk/src/BDTheque/DataModelBundle/Entity/AuteurSerie.php
<?php

namespace BDTheque\DataModelBundle\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * BDTheque\DataModelBundle\Entity\AuteurSerie
 *
 * @ORM\Table(
 *      name="series_auteurs", 
 *      uniqueConstraints={
 *          @ORM\UniqueConstraint(name="auteur_idx", columns={"personne_id","serie_id","metier"})
 *      })
 * @ORM\Entity
 * @ORM\InheritanceType("SINGLE_TABLE")
 * @ORM\DiscriminatorColumn(name="metier", type="string", length=1)
 * @ORM\DiscriminatorMap({"S" = "ScenaristeSerie", "D" = "DessinateurSerie", "C" = "ColoristeSerie"})
 */
abstract class AuteurSerie extends Auteur {

    /**
     * @var Serie $serie
     *
     * @ORM\ManyToOne(targetEntity="Serie", cascade={"all"}, fetch="LAZY")
     * @ORM\JoinColumn(nullable=false)
     */
    private $serie;

    public function __construct(Serie $serie, Personne $personne) {
        $this->serie = $serie;
        $this->setPersonne($personne);
    }

    /**
     * Set serie
     *
     * @param Serie $serie
     * @return AuteurSerie
     */
    public function setSerie(Serie $serie) {
        $this->serie = $serie;

        return $this;
    }

    /**
     * Get serie
     *
     * @return Serie
     */
    public function getSerie() {
        return $this->serie;
    }

}

?>